Update develop-in-the-cloud.md (#2988)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/develop-in-the-cloud.md

---

> title: "dbt Cloud IDE""

Author: ```suggestion
title: "dbt Cloud IDE"
```

> title: "dbt Cloud IDE""

Author: ```suggestion
title: "dbt Cloud IDE"
```

With the Cloud IDE, you can:

- Write modular SQL models with select statements and the [ref()](/reference/dbt-jinja-functions/ref) function,
- Compile dbt code into SQL and execute it against your database directly,
- Test every model before deploying them to production,
- Generate and view documentation of your dbt project,
- Leverage git and version-control your code from your browser with a couple of clicks,
- Create and test Python models:
    * Compile Python models to see the full function that gets executed in your data platform
    * See Python models in DAG in dbt version 1.3 and higher
    * Currently, you can't preview python models
- Visualize a directed acyclic graph (DAG), and [more](#cloud-ide-features).

<Lightbox src src="/img/docs/dbt-cloud/cloud-ide/cloud-ide-v2.jpg" width="85%" title="The dbt Cloud IDE with all its features is visible in dark mode."/>

## Prerequisites

To develop in the Cloud IDE, make sure you have the following:

> - A [dbt Cloud account](https://cloud.getdbt.com/) and [Developer seat license](/docs/collaborate/manage-access/seats-and-users)

Author: ```suggestion
- A [dbt Cloud account](https://cloud.getdbt.com/) and [Developer seat license](/docs/cloud/manage-access/seats-and-users)
```

> - A git repository set up and git provider must have `write` access enabled. See [Connecting your GitHub Account](/docs/collaborate/git/connect-github) or [Importing a project by git URL](/docs/collaborate/git/import-a-project-by-git-url) for detailed setup instructions

Author: ```suggestion
- A git repository set up and git provider must have `write` access enabled. See [Connecting your GitHub Account](/docs/cloud/git/connect-github) or [Importing a project by git URL](/docs/cloud/git/import-a-project-by-git-url) for detailed setup instructions
```

> - A dbt project connected to a [data platform](/docs/get-started/connect-your-database)

Author: ```suggestion
- A dbt project connected to a [data platform](/docs/cloud/connect-data-platform/connect-your-database)
```

> - Set up a [development environment and development credentials](#access-the-cloud-ide) on dbt version 1.0 or higher

Author: ```suggestion
- Set up a [development environment and development credentials](#access-the-cloud-ide)
- The environment must be on dbt version 1.0 or higher
```

| Feature  |  Info |
|---|---|
| **Keyboard shortcuts** | You can access a variety of [commands and actions](/docs/cloud/dbt-cloud-tips#cloud-ide-keyboard-shortcuts) in the IDE by choosing the appropriate keyboard shortcut. This includes the command palette (Command-P or Control-P), which has common shortcuts and build commands to optimize your developer workflow. For example, you can build modified models or continue the build from the last failure point. |
| **File state indicators**  |  Ability to see when changes or actions have been made to the file. The indicators **M, D, A,** and **•** appear to the right of your file or folder name and indicate the actions performed: <br /> <br /> - Unsaved **(•)** &mdash; The IDE detects unsaved changes to your file/folder<br /> - Modification **(M)** &mdash; The IDE detects a modification of existing files/folders<br /> - Added **(A)** &mdash; The IDE detects added files<br/> - Deleted **(D)** &mdash; The IDE detects deleted files.

> | **IDE version control** | The IDE version control section and git button allows you to apply the concept of [version control](/docs/collaborate/git/version-control-basics) to your project directly into the IDE. <br /><br /> - Create or change branches<br /> - Commit or revert individual files by right-clicking the edited file<br /> - [Resolve merge conflicts](/docs/collaborate/git/merge-conflicts)<br /> - Execute git commands using the git button <br /> - Link to the repo directly by clicking the branch name |

Reviewer: ```suggestion
| **IDE version control** | The IDE version control section and git button allow you to apply the concept of [version control](/docs/collaborate/git/version-control-basics) to your project directly into the IDE. <br /><br /> - Create or change branches<br /> - Commit or revert individual files by right-clicking the edited file<br /> - [Resolve merge conflicts](/docs/collaborate/git/merge-conflicts)<br /> - Execute git commands using the git button <br /> - Link to the repo directly by clicking the branch name |
```

## Cloud IDE features

> The dbt Cloud IDE comes with [tips](/docs/get-started/dbt-cloud-tips) and features that make it easier for you to develop, build, compile, run, and test data models. 

Author: ```suggestion
The dbt Cloud IDE comes with [tips](/docs/cloud/dbt-cloud-tips) and features that make it easier for you to develop, build, compile, run, and test data models. 
```

| Feature  |  Info |
|---|---|

> | **Keyboard shortcuts** | You can access a variety of [commands and actions](/docs/get-started/dbt-cloud-tips#cloud-ide-keyboard-shortcuts) in the IDE by choosing the appropriate keyboard shortcut. This includes the command palette (Command-P or Control-P), which has common shortcuts and build commands to optimize your developer workflow. For example, you can build modified models or continue the build from the last failure point. |

Author: ```suggestion
| **Keyboard shortcuts** | You can access a variety of [commands and actions](/docs/cloud/dbt-cloud-tips#cloud-ide-keyboard-shortcuts) in the IDE by choosing the appropriate keyboard shortcut. This includes the command palette (Command-P or Control-P), which has common shortcuts and build commands to optimize your developer workflow. For example, you can build modified models or continue the build from the last failure point. |
```

Set up your developer credentials:

> 1. Navigate to your **Credentials** under **Your Profile** settings, which you can access at `https://YOUR_ACCESS_URL/settings/profile#credentials`, replacing `YOUR_ACCESS_URL` with the [appropriate Access URL](/docs/deploy/regions-ip-addresses) for your region and plan.

Author: ```suggestion
1. Navigate to your **Credentials** under **Your Profile** settings, which you can access at `https://YOUR_ACCESS_URL/settings/profile#credentials`, replacing `YOUR_ACCESS_URL` with the [appropriate Access URL](/docs/cloud/about-cloud/regions-ip-addresses) for your region and plan.
```

## Related docs

> - [dbt Cloud features](/docs/get-started/dbt-cloud-features)

Author: ```suggestion
- [dbt Cloud features](/docs/cloud/about-cloud/dbt-cloud-features)
```
